shaman/controllers/nodes/__init__.py

```diff
@@ -42,6 +43,14 @@ def index(self):
             resp[node.host] = node.__json__()
         return resp
 
+    @secure(basic_auth)
+    @expose(method='GET', content_type="text/plain")
+    def next(self):
+        next_node = get_next_node()
+        if not next_node:
+            abort(500, "There are no healthy chacra nodes available.")
```

Comment:
If we go strictly by the spec, a `500` would be raised because the *"server encountered an unexpected condition which prevented it from fulfilling the request"*

But this is not unexpected, the server knows and is correctly handling the request. Some expectations on `404` define it as: *"The requested resource could not be found but may be available again in the future. Subsequent requests by the client are permissible."* which sounds correct in this case